covid19-dashboard/dc-covid19/src/DataCalculator.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ Copyright 2020 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import {addOrSubtractNDaysToDate, getRealISODatesFromArrayOfDeltaDays} from './Utils'
+
+type dataPerGeoIdPerDate = {string: {string: number}} | {}
+type calculationFunction = (val1: number, val2: number, assistanceObject: string) => number
+
+/**
+ * Performs a calculation() iteratively on two dates.
+ * @param data: input data in the form of dates->geoIds->value.
+ * @param range: the range of dates to calculate the data for in ISO Format. Example: ["2020-01-01", "2020-01-10"]
+ * @param deltaDays: Do we wanna compare the data every 2 days? every 7? Any number works.
+ * @param calculation: The function in charge of doing some calculation on the data for dates X and Y.
+ * @private
+ */
+const _performIterationCalculation = (data: dataPerGeoIdPerDate, range: [string, string],
```

Comment:
This is a re-used method. All calculations use this method.
It iterates through all the dates and all the geoIds and returns only the dates in a specific range.

**It takes a calculation function as a parameter.**